File: backend/src/main/java/com/apocalypse/thefall/dto/item/instance/ArmorInstanceDto.java
package com.apocalypse.thefall.dto.item.instance;

import com.apocalypse.thefall.dto.item.ItemDto;
import com.apocalypse.thefall.entity.item.enums.EquippedSlot;

public record ArmorInstanceDto(
        Long id,
        EquippedSlot equippedSlot,
        ItemDto item) {
}

File: backend/src/main/java/com/apocalypse/thefall/dto/item/instance/ArmorInstanceDto.kt
package com.apocalypse.thefall.dto.item.instance

import com.apocalypse.thefall.dto.item.ItemDto
import com.apocalypse.thefall.entity.item.enums.EquippedSlot

@JvmRecord
data class ArmorInstanceDto(
    override val id: Long?,
    val equippedSlot: EquippedSlot?,
    override val item: ItemDto?
) : ItemInstanceDto 